mango/dashboards/file_explorer_app.py

```diff
@@ -504,12 +515,26 @@ def _render_file_content(self, path_selected: str, key: str):
         elif path_selected.endswith(".html"):
             with st.spinner("Wait for it..."):
                 try:
-                    with open(path_selected, "r") as f:
-                        components.html(
-                            f.read(),
+                    try:
+                        fig = self._read_plot_from_html(path_selected)
+                        fig.update_layout(
                             height=self.config.get(f"height_{key}", 500),
                             width=self.config.get(f"width_{key}", None),
                         )
+                        st.plotly_chart(
+                            fig,
+                            use_container_width=True
+                            if self.config.get(f"width_{key}", None) is None
+                            else False,
+                            theme=None,
+                        )
+                    except Exception:
+                        with open(path_selected, "r") as f:
+                            components.html(
+                                f.read(),
+                                height=self.config.get(f"height_{key}", 500),
+                                width=self.config.get(f"width_{key}", None),
+                            )
                 except Exception as e:
                     try:
                         with open(path_selected, "r", encoding="utf8") as f:
```

Comment:
Why do we have a try inside a another try and then a try inside the except clause of the top most try?
This seems like a hack that should not be done and should not have been merged. There has to be a better way to handle errors in this part of the code

Reply:
Okay, let me change this part of the code and correct the error handling in a more efficient way.